src/app/components/SearchInput/index.tsx

```diff
@@ -0,0 +1,37 @@
+import * as React from 'react';
+// import * as style from './style.css';
+
+export namespace _SearchInput {
+  export interface Props {
+    value: string;
+    className?: string;
+    placeHolder?: string;
+    inputStyle?: string;
+
+    handleKeyChange: (value: string) => void;
+  }
+}
+
+export const SearchInput: React.FC<_SearchInput.Props> = (props: _SearchInput.Props) => {
+  const [inputValue, setInputValue] = React.useState<string>(String(props.value));
```

Comment:
This is maybe overkill to ensure it's a string. Was the namespace necessary? Or just a preference?

Reply:
Oh is that so? I'll check that one. For the namespace, I'm just more familiar with that structure when we used redux before.

Reply:
React.FC is not the best practice for function component. 
For more details
https://github.com/typescript-cheatsheets/react